IGC/Compiler/Optimizer/OpenCLPasses/WIFuncs/WIFuncResolution.cpp

```diff
@@ -303,7 +303,7 @@ static Value* BuildLoadInst(CallInst& CI, unsigned int Offset, Type* DataType)
     auto Size = ElemByteSize;
     if (DataType->isVectorTy())
     {
-        Size *= DataType->getVectorNumElements();
+        Size *= cast<IGCLLVM::FixedVectorType>(DataType)->getNumElements();
     }
```

Comment:
This could be further simplified as such:
```c++
if (auto DataVT = dyn_cast<IGCLLVM::FixedVectorType>(DataType))
{
    Size *= DataVT->getNumElements();
}
```
but a case can be made it's less readable, so let's leave it as is.